### PR TITLE
Set JAVA_HOME when running closure compiler

### LINKED ISSUE #10304: Closure compiler breakage on windows emscripten-releases builder
Since #9989 the emscripten-releases builder has been unable to roll the emscripten version.

The logs looks like this:

```
Error: EPERM: operation not permitted, mkdir 'C:\'
    at Object.mkdirSync (fs.js:806:3)
    at C:\b\s\w\ir\k\install\emscripten\node_modules\google-closure-compiler\cli.js:32:10
    at Array.reduce (<anonymous>)
    at mkDirByPathSync (C:\b\s\w\ir\k\install\emscripten\node_modules\google-closure-compiler\cli.js:29:24)
    at C:\b\s\w\ir\k\install\emscripten\node_modules\google-closure-compiler\cli.js:179:13
    at Array.forEach (<anonymous>)
    at C:\b\s\w\ir\k\install\emscripten\node_modules\google-closure-compiler\cli.js:178:32 {
  errno: -4048,
  syscall: 'mkdir',
  code: 'EPERM',
  path: 'C:\\'
}
```

https://chromium-review.googlesource.com/c/emscripten-releases/+/2026425

I tracked the problem down to a chain of different issues:

1. The emscripten-releases builder doesn't install the native version of the compiler (most likely because it doesn't find a compiler it likes at npm install time).
2. The default closure compiler decides to choose the JS version even thought java version is installed (https://github.com/google/closure-compiler-npm/issues/160)
3. The JS version seems to broken on windows which is the actual error we see (https://github.com/google/closure-compiler-npm/issues/161)

### PR DESCRIPTION
**Pull request by sbc100**
This allows the closure compiler to successfully fall back to the
java version if the native version is not found.

See: https://github.com/google/closure-compiler-npm/issues/160

Fixes #10304

### CODE REVIEW

**Comment by juj** [comment]
Can we only apply this forced java if user did not pass a `--closure-args --platform=javascript` flag? We are not seeing these issues in our own testing, so we'd like to be able to use the JS version, perhaps explicitly passing `--closure-args --platform=javascript` if we don't like that on by default?

**Comment by sbc100** [comment]
Are you sure you are not using the native version rather than the JS version?  It seems the native version is always the default internally.   I tried the JS version and it was super slow.

But yes I think we should do as you say. 

I'd also like to find a way to allow the native version to continue to take precedence.

**Comment by juj** [comment]
> I'd also like to find a way to allow the native version to continue to take precedence.

Oh right, actually when I wrote above comment I forgot the existence of the native version altogether and thought it was the JS version I've been using, yeah, we'd then be using `--platform=native` with precedence instead of `--platform=javascript`.

**Comment by sbc100** [comment]
Ok, I found a better solution which is simply to set JAVA_HOME.  This allows the java backend to be selected in the absence of the native backend.

**Comment by sbc100** [comment]
PTAL ASAP, I'm hoping this change will unblock out release builder and enable us to cut a release.

File: tools/shared.py [hunk 1]
```diff
@@ -2534,6 +2534,11 @@ def closure_compiler(filename, pretty=True, advanced=True, extra_closure_args=[]
       logger.debug('closure compiler: ' + ' '.join(args))
       env = os.environ.copy()
       env['PATH'] = env['PATH'] + os.pathsep + get_node_directory()
+      # Closure compiler expects JAVA_HOME to be set in order to enable the java backend.  Without
+      # this is will only try the native and javascriptn version.
```
Comment (kripken) [inline tools/shared.py]: typo in `javascript`, extra `n` at the end

**Pull request merged**